### DISCUSSION TITLE
Suspend importing while playing

### DISCUSSION THREAD

**Discussion by Koltay** [ppy/osu#19282] [Ideas]
To improve playing stability, suspending any importing during playing could fix the issue.

I have ~10k scores in stable and when I clicked "Import Scores" and started playing in lazer, it stuttered massively and I could not play normally;

Adding the function to automatically suspend any tasks during gameplay would help a lot.

(Furthermore, it may also apply to downloading large amounts of beatmaps as well)

![image](https://user-images.githubusercontent.com/73950784/180258244-acf97dfc-7deb-4a60-b38d-04c021b1d739.png)

![image](https://user-images.githubusercontent.com/73950784/180258516-424f0b69-ab34-4991-bf6c-6785d235ec02.png)

**Comment by peppy** [comment]
That sounds weird, score imports shouldn't really cause the game to grind to a halt. Would be interested in collecting more performance data (starting with log files I guess).

**Comment by Koltay** [comment]
I think database.log should be the log you wanted?
database.log: https://puu.sh/Jcfcf/7a4f72cf90.log

(Edit: and network.log: https://puu.sh/Jcfde/aed9aed799.log)

**Reply by peppy** [reply]
Thanks.

I wonder if you'd also be willing to provide your scores in a zip file. This would help me attempt to reproduce this locally.

**Reply by Koltay** [reply]
Sure. Here is the zip file of my scores at osu!\data\r: 
https://drive.google.com/file/d/1aFR6zy4W13pxTwWjFmWgqzGDDsPYUaeQ/view?usp=sharing

Moreover, I forgot to let you know that there are errors during the import: 
![image](https://user-images.githubusercontent.com/73950784/180994243-98bb6d22-374a-4c7e-8e68-09755c5dbbea.png)
Hope it helps during debugging
